src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let stateCopy = { ...state };
+  const statesOfCopy = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy = { ...stateCopy }, action.extraData);
```

Comment:
You have unnecessary assignment here.
`Object.assign(stateCopy, action.extraData)`  would be better in my opinion, less code and operations. 
And with that you can declare your `stateCopy` variable with `const`